src/ethereum/shanghai/fork.py

```diff
@@ -529,6 +550,12 @@ def apply_body(
         if account_exists_and_is_empty(state, wd.address):
             destroy_account(state, wd.address)
 
+    # If any inclusion list addresses are not satisfied, the block is invalid.
+    for addr in inclusion_list_nonces:
+        sender_account = get_account(env.state, addr)
+        if inclusion_list_nonces[addr] < sender_account.nonce:
+            raise InvalidBlock
+
```

Comment:
If we remove 'nonce` for spec, this could be validated before any transaction execution!

Reply:
as mentioned above, the nonce is for the transactions that could appear in the parent block